lib/top_secret/text.rb

```diff
@@ -16,7 +16,7 @@ def initialize(input, custom_filters: [], filters: {}, model: nil)
       @output = input.dup
       @mapping = {}
 
-      @model = model || Mitie::NER.new(TopSecret.model_path)
+      @model = (model || TopSecret.model_path) ? Mitie::NER.new(TopSecret.model_path) : nil
```

Comment:
I don't think this works as expected. If we were to pass a new model during initialization, we'd end up calling `Mitie::NER.new(TopSecret.model_path)`, end not using the model that was passed.

It just so happens that the only model we pass in is an instance of `Mitie::NER`.